### DISCUSSION TITLE
ScreenElement collider error, and ScreenElement works only for graphics?

### DISCUSSION THREAD

**Discussion by KokoDoko** [excaliburjs/Excalibur#2597] [Q&A] [answered]
I am using the ScreenElement for a UI, so that it always remains in place when the camera is moving around.
But I keep getting this error as soon as I add a ScreenElement to the game - even if the ScreenElement is completely empty.

```
Excalibur only supports convex polygon colliders and will not behave properly.Call PolygonCollider.triangulate() to build a new collider composed of smaller convex triangles
```

Code

```typescript
class Game extends Engine {

    constructor() {
        super({ width: 800, height: 600 })
        Physics.useRealisticPhysics()
        Physics.gravity = new Vector(0, 800)
        const loader = new Loader([...])
        this.start(loader).then(() => this.startGame())
    }

    startGame(){
        let test = new ScreenElement()
        this.add(test)
    }
}
new Game()
```

Another problem is that if I use `addChild` to add anything that is not a graphic, then that child still responds to the camera.

**Comment by eonarheim** [comment, marked as answer]
Hi @KokoDoko that error also seems like a bug... I think it might be because there is no width/height specified in the ScreenElement so it tries to create a 0/0 collider causing the message. Thanks for spotting, I'll add another bug for that!

To have the children independent of the camera try setting their [`TransformComponent.coordPlan`](https://excaliburjs.com/docs/api/edge/classes/TransformComponent.html#coordPlane) to [Screen](https://excaliburjs.com/docs/api/edge/enums/CoordPlane.html#Screen) or use another `ScreenElement` as the child.

`ScreenElement` under the hood is really [just sugar](https://github.com/excaliburjs/Excalibur/blob/main/src/engine/ScreenElement.ts) on top of actor to set a couple of properties that are convenient for that use case.

**Reply by eonarheim** [reply]
@KokoDoko I've created the issue https://github.com/excaliburjs/Excalibur/issues/2601 👍 

Thanks again for bringing this up!

**Reply by eonarheim** [reply]
This should be fixed in the latest alpha build soon 👍
